agent/doc/UML/hydra_lab_exception_monitor_design.puml

```diff
@@ -0,0 +1,56 @@
+@startuml exception_monitor
+interface ExceptionReporter {
+    + void reportException(Exception e)
+    + void reportException(String message, StackTraceElement[] stackTrace)
```

Comment:
```suggestion
    + void reportException(String message, Exception e, Thread threadContext)
```

Reply:
When you have the thread object as context, it's easy to get all the stacktraces.